### PR TITLE
GKE Security Posture configuration fixes

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
This fixes the main configurations as well as a few examples. One of the prominent examples I have not covered here is the node-exporter because that has a few issues and I am not familiar with the exporter enough to determine if it would break anything. To be more specific, they have issues with using the host network and volume mounts which I don't believe we could remove.

### CODE REVIEW

File: cmd/operator/deploy/operator/11-rule-evaluator.yaml [hunk 1]
```diff
@@ -49,11 +49,19 @@ spec:
                 operator: In
                 values:
                 - linux
-      serviceAccountName: collector
```
Comment (pintohutch) [inline cmd/operator/deploy/operator/11-rule-evaluator.yaml:52]: You removed this as it's [already part](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/c4f3239d50e536b60ba730e91fce1f9fa3069c93/cmd/operator/deploy/operator/11-rule-evaluator.yaml#L52) of the template spec, correct?

File: examples/node-exporter.yaml [hunk 2]
```diff
@@ -69,6 +69,15 @@ spec:
           requests:
             cpu: 102m
             memory: 180Mi
+        securityContext:
```
Comment (pintohutch) [inline examples/node-exporter.yaml:72]: With the exception of the `frontend`, I'm tempted to leave the examples directory out of this change for now.

They're not officially a part of the managed infrastructure and I'd rather not risk breaking anything which our [docs](https://cloud.google.com/stackdriver/docs/managed-prometheus/exporters/node_exporter) point to.

**Review by pintohutch — COMMENTED** [review]
Nice! It looks like it was mostly `initContainers` anyway.

I'd say the only other thing to try out - maybe it would be simpler to consolidate `securityContext` at the [pod-leve](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/c4f3239d50e536b60ba730e91fce1f9fa3069c93/cmd/operator/deploy/operator/05-deployment.yaml#L96-L98) instead of the container level?

Though that could always be done later, so no strong opinion.

File: cmd/operator/deploy/operator/11-rule-evaluator.yaml (hunk 1 above)
Comment (TheSpiritXIII) [reply cmd/operator/deploy/operator/11-rule-evaluator.yaml:52]: Yes.

**Review by pintohutch — APPROVED** [review]
LGTM!

**Pull request merged**